auth/auth0_manager.go

```diff
@@ -1,23 +1,95 @@
 package auth
 
 import (
-	"log"
+	"fmt"
+	"time"
 
+	"github.com/zpatrick/go-cache"
 	"github.com/zpatrick/rclient"
 )
 
 type Auth0Manager struct {
-	token  string
-	client *rclient.RestClient
+	clientID   string
+	connection string
+	client     *rclient.RestClient
+	cache      *cache.Cache
 }
 
-func NewAuth0Manager(endpoint, token string) *Auth0Manager {
+type Auth0Config struct {
+	Domain     string
+	ClientID   string
+	Connection string
+}
+
+type oauthReq struct {
+	ClientID   string `json:"client_id"`
+	Connection string `json:"connection"`
+	Username   string `json:"username"`
+	Password   string `json:"password"`
+	GrantType  string `json:"grant_type"`
+	Scope      string `json:"scope"`
+}
+
+type authStatus struct {
+	isValid bool
+	penalty time.Duration
+}
+
+const (
+	maxPenalty      = 5 * time.Second
+	validAuthExpiry = 1 * time.Hour
+)
+
+func NewAuth0Manager(config Auth0Config) *Auth0Manager {
```

Comment:
Fair enough - I was thinking that it looked slightly nicer in `main.go` to call with a config object than with `c.String("sadfasdf")` three times.

✅